### PR TITLE
Fix flooding of map. Text on path

### PR DESCRIPTION
**Pull request by ivanPyrohivskyi**
Fixed draw text on path using Harfbuzz.
Added correlation between Harfbuzz and Skia witdths. Fixed centering of the first glyphs.

### CODE REVIEW

File: native/src/textdraw.cpp [hunk 1]
```diff
@@ -920,32 +920,54 @@ void FontRegistry::drawHbTextOnPath(SkCanvas *canvas, std::string textS, SkPath
 	hb_buffer_destroy(hb_buffer);
 	hb_font_destroy(hb_font);
 
+	SkPathMeasure meas(path, false);
+	// check correlation between harfbuzz and skia
+	if (xy[length - 1].x() > meas.getLength()) 
+	{
```
Comment (vshcherb) [inline native/src/textdraw.cpp:926]: core-legacy C++ has different code style (check Google code style

File: native/src/textdraw.cpp [hunk 2]
```diff
@@ -920,32 +920,54 @@ void FontRegistry::drawHbTextOnPath(SkCanvas *canvas, std::string textS, SkPath
...
+		SkScalar correlation = meas.getLength() / xy[length - 1].x();
+		for (int i = 0; i < length; ++i)
+		{
+			xy[i].set(xy[i].x() * correlation, xy[i].y());
```
Comment (vshcherb) [inline native/src/textdraw.cpp:930]: if correlation = 0.1, does it mean glyphs will go over each other?

Comment (ivanPyrohivskyi) [reply native/src/textdraw.cpp:936]: This is unlikely but I added check correlation less than half of measure lenght.

File: native/src/textdraw.cpp (hunk 1 above)
Comment (ivanPyrohivskyi) [reply native/src/textdraw.cpp]: Ok

**Pull request merged**